ducc/lib/utils.go

```diff
@@ -1,19 +1,108 @@
 package lib
 
 import (
+	"bytes"
+	"encoding/base32"
+	"encoding/base64"
+	"io"
 	"io/ioutil"
+	"os"
 	"path"
 	"strings"
+
+	"github.com/google/uuid"
 )
 
 // this flag is populated in the main `rootCmd` (cmd/root.go)
 var (
 	TemporaryBaseDir string
 )
 
+//function that satisfies io.Closer.
+type CloseFunc func() error
+
+type readCloser struct {
+	io.Reader
+	c CloseFunc
+}
+
+func (r readCloser) Close() error {
+	return r.c()
+}
+
+//wrapper around bytes.Buffer which implements close methods too.
+type ReadCloserBuffer struct {
+	*bytes.Buffer
+}
+
+func (cb ReadCloserBuffer) Close() (err error) {
+	return
+}
+
 func UserDefinedTempDir(dir, prefix string) (name string, err error) {
 	if strings.HasPrefix(dir, TemporaryBaseDir) {
 		return ioutil.TempDir(dir, prefix)
 	}
 	return ioutil.TempDir(path.Join(TemporaryBaseDir, dir), prefix)
 }
+
+// TeeReadCloser returns a io.ReadCloser that writes everything it reads from r to w.
+// The Close method for the returned ReadCloser is same as that of r.
+func TeeReadCloser(r io.ReadCloser, w io.Writer) io.ReadCloser {
+	return readCloser{
+		io.TeeReader(r, w),
+		r.Close,
+	}
+}
+
+//Trim the digest algorithm name from digest
+func calculateId(digest string) string {
+	return strings.Split(digest, ":")[1]
+}
+
+//generates the file name for link dir in podman store
+func generateID(l int) string {
+	id := uuid.New()
```

Comment:
There is no implementation of this in standard lib. We can generate one using rand.Reader.
Like here: https://play.golang.org/p/4FkNSiUDMg
But isn't using a library better?

Reply:
Well the implementation in github.com/google/uuid is also similar to above code snippet. Since we don't any other library feature, I think we can go ahead without it!

Reply:
I don't believe we need a whole UUID, but just some reasonably random characters. If we can avoid pulling in another library it would be better.

Reply:
Done